File: lista2/src/Kwadrat.cpp
#include "funs.hpp"

// Kwadrat::Kwadrat(double a, double b, double c, double d, double kat) {
    
// }
// Kwadrat::Kwadrat(double a) : Czworokat(a,a,a,a,90) {
//     //this-> a = a;
// }

double Kwadrat::pole() {
    return getA() * getA();
}
double Kwadrat::obwod() {
    return 4*getA();
}

File: Lista2 java/src/Kwadrat.java
public class Kwadrat extends Czworokat{

	public Kwadrat(double a) throws BlednaWartoscKataException {
		super(a, a,a,a,90);
	}

	@Override
	public double pole() {
		return getA()*getA();
	}

	@Override
	public double obwod() {
		return super.obwod();
	}

	@Override
	public String toString() {
		return "Kwadrat";
	}
}
